src/internal_events/codecs.rs

```diff
@@ -126,10 +127,12 @@ impl<E: std::fmt::Display> InternalEvent for EncoderWriteAllError<'_, E> {
             "error_type" => error_type::ENCODER_FAILED,
             "stage" => error_stage::SENDING,
         );
-        emit!(ComponentEventsDropped {
-            count: 1,
-            intentional: false,
-            reason,
-        });
+        if self.count > 0 {
```

Comment:
I'm curious when we'd be attempting to write 0 things 🤔

Reply:
My goal was to make this event more flexible per our prior discussion.
In that vein, one could emit this event with count = 0 to not fire the events dropped event.

Reply:
But maybe all write failures in the encoder would fall under the category of dropping events ?

Reply:
Fair - I remembered a PR from @prognant that ended up dropping the a count check 👍

Reply:
Given the callers emitting `EncoderWriteError` are now utilizing the case of count = 0, marking this as resolved.